### PR TITLE
Onboarding Step 1

### PR DESCRIPTION
**Pull request by dakota002**
Adds legacy_users table, checks if user is a legacy user if they do not have valid permissions, updates sandbox seed and invoker tests

Should address the 2nd and 3rd points of https://github.com/nasa-gcn/gcn.nasa.gov/issues/291
Point 1 will likely be a one off script, once the other steps are set up.
Point 4 might need to be wrapped into the post sign up confirm lambda

### CODE REVIEW

File: app/email-incoming/index.ts [hunk 1]
```diff
@@ -87,12 +92,19 @@ async function handleRecord(record: SNSEventRecord) {
...
-  }
+  const userData = userTypeData
+    ? {
+        sub: extractAttributeRequired(userTypeData, 'sub'),
+        email: extractAttributeRequired(userTypeData, 'email'),
+        name: extractAttribute(userTypeData, 'name'),
+        affiliation: extractAttribute(userTypeData, 'custom:affiliation'),
+      }
+    : {
+        sub: legacyUserData.email,
```
Comment (lpsinger) [inline app/email-incoming/index.ts:103]: This is not an OIDC subject.

```suggestion
```

**Review by lpsinger — CHANGES_REQUESTED** [review]
I suggest restructuring this as two functions, `getUser` and `getLegacyUser`, both of which return a promise of a type like this:

```ts
interface UserData {
  email: string
  sub?: string
  name?: string
  affiliation?: string
}
```

And then you can get the user by using the nullish coalescing operator:

```ts
const user = (await getUser()) ?? (await getLegacyUser())
```

File: app/email-incoming/index.ts (hunk 1 above)
Comment (jracusin) [reply app/email-incoming/index.ts:103]: Make sure you track that some legacy users are submit & receive, or only submit, or only receive.

File: app/email-incoming/index.ts [hunk 2]
```diff
@@ -133,10 +118,47 @@ export async function handler(event: SNSEvent) {
...
+    (user) => extractAttributeRequired(user, 'email') == userEmail
+  )
+  return userTypeData
+    ? {
+        sub: extractAttributeRequired(userTypeData, 'sub'),
+        email: extractAttributeRequired(userTypeData, 'email'),
+        name: extractAttribute(userTypeData, 'name'),
+        affiliation: extractAttribute(userTypeData, 'custom:affiliation'),
+      }
+    : undefined
```
Comment (lpsinger) [inline app/email-incoming/index.ts:144]: Two slightly shorter options:

```suggestion
  if (userTypeData)
    return {
        sub: extractAttributeRequired(userTypeData, 'sub'),
        email: extractAttributeRequired(userTypeData, 'email'),
        name: extractAttribute(userTypeData, 'name'),
        affiliation: extractAttribute(userTypeData, 'custom:affiliation'),
      }
```

or

```suggestion
  return userTypeData && {
      sub: extractAttributeRequired(userTypeData, 'sub'),
      email: extractAttributeRequired(userTypeData, 'email'),
      name: extractAttribute(userTypeData, 'name'),
      affiliation: extractAttribute(userTypeData, 'custom:affiliation'),
    }
```

File: app/email-incoming/index.ts (hunk 1 above)
Comment (dakota002) [reply app/email-incoming/index.ts]: Sweet, okay I updated it to use two functions, but opted for names getCognitoUserData (so as not to confuse with the existing getUser function) and getLegacyUserData. 

As for the receive/submit difference, that should all be okay, for this stuff we are specifically focusing on Submitters, so the only users that should get uploaded into this specific table should be Submitters

File: app/email-incoming/index.ts [hunk 3]
```diff
@@ -133,10 +118,47 @@ export async function handler(event: SNSEvent) {
...
   const db = await tables()
-  return await db.legacy_users.get({ email: userEmail })
+  const data = await db.legacy_users.get({ email: userEmail })
+  return data
+    ? {
+        email: data.email,
+        name: data.name,
+        affiliation: data.affiliation,
+      }
+    : undefined
```
Comment (lpsinger) [inline app/email-incoming/index.ts:163]: (same comment as above)

File: app/lib/email.ts [hunk 4]
```diff
@@ -45,7 +46,11 @@ export async function sendEmail(
   try {
     await client.send(command)
   } catch (e) {
-    if (
+    if (e instanceof MessageRejected && hostname == 'localhost') {
```
Comment (lpsinger) [inline app/lib/email.ts:49]: This must still throw in production.

**Review by lpsinger — CHANGES_REQUESTED** [review]

File: app/lib/email.ts (hunk 4 above)
Comment (dakota002) [reply app/lib/email.ts:49]: That should be okay though since the the hostname should be different right?

Comment (dakota002) [reply app/lib/email.ts:49]: Hmm true, think I should just let it throw normally and remove this condition?

Comment (lpsinger) [reply app/lib/email.ts:49]: That's fine with me.

File: app/email-incoming/index.ts (hunk 1 above)
Comment (lpsinger) [reply app/email-incoming/index.ts]: Whether it's the same table or a different one, we just have to remember to take care of this. I've made a note of it at https://github.com/nasa-gcn/gcn.nasa.gov/issues/612#issuecomment-1433651046.

**Pull request merged**